app/views.py

```diff
@@ -0,0 +1,242 @@
+            })
+            if req_ser.is_valid():
+                req_ser.save()
+                return Response({"message":"Sent notificaion", "Request":req_ser.data}, status=status.HTTP_200_OK)
+            else:
+                return Response({"message":req_ser.errors}, status=status.HTTP_400_BAD_REQUEST)
+        except Exception as e:
+            return Response({"message":str(e)}, status=status.HTTP_400_BAD_REQUEST)
+
+
+class ViewRequests(APIView):
+    permission_classes = (IsAuthenticated,)
+
+    def get(self, request):
+        user = request.user
+        today_date = datetime.today().strftime('%Y-%m-%d')
+        req_objects = Requests.objects.filter(Q(date_time_creation__date=today_date) & ~Q(user_id=user.id))
+        response = RequestsSerializer(req_objects, many=True)
+        resp = response.data
+        for req in resp:
+            user_req = User.objects.get(id=req['user_id'])
+            req['user_username'] = user_req.username
+        return Response({"message":"Received Requests", "Requests":resp}, status=status.HTTP_200_OK)
+
+
+class ViewChatRooms(APIView):
+    permission_classes = (IsAuthenticated,)
+
+    def get(self, request):
+        user = request.user
+        chatRooms = ChatRoom.objects.filter( Q(participant1_id=user.id) | Q(participant2_id=user.id))
+        chatRooms_serializer = ChatRoomSerializer(chatRooms, many=True)
+        resp = chatRooms_serializer.data
+        if len(resp) == 0:
+            return Response({"message":"No chat rooms available"}, status=status.HTTP_204_NO_CONTENT)
+        else:
+            resp = chatRooms_serializer.data
+            for chatroom in resp:
+                user1 = User.objects.get(id=chatroom['participant1_id'])
+                user2 = User.objects.get(id=chatroom['participant2_id'])
+                chatroom['participant1_username'] = user1.username
+                chatroom['participant2_username'] = user2.username
+            return Response({"message":"Chat rooms found", "ChatRooms":resp}, status=status.HTTP_200_OK)
+
+class ViewUserDetails(APIView):
+    permission_classes = (IsAuthenticated,)
+
+    def get(self, request):
+        user = request.user
+        return Response({
+            "message":"User Details",
+            "User": {
+                "id": user.id,
+                "email":user.email,
+                "username":user.username,
+                "phone_no":user.phone_no,
+                "date_of_birth":user.date_of_birth
+            }                
+        })
+
+
+class PreviousMessagesView(APIView):
+    permission_classes = (IsAuthenticated,)
+
+    def get(self, request, pk):
+        user = request.user
+        chatroom = ChatRoom.objects.filter(Q(participant1_id=user.id) | Q(participant2_id=user.id))
+        if len(chatroom) == 0:
+            return Response({"message":"Invalid Chat Room"}, status=status.HTTP_400_BAD_REQUEST)
+        messages = Messages.objects.filter(chat_room_id=pk)[:30]
```

Comment:
why is there a limit(30) to the amount of messages you are fetching.

Reply:
Just to have a threshold for the number of last messages that a user can see.